addon/object/index.js

```diff
@@ -55,8 +55,10 @@ export const action = decorator(function(target, key, desc) {
   // We must collapse the superclass prototype to make sure that the `actions`
   // object will exist. Since collapsing doesn't generally happen until a class is
   // instantiated, we have to do it manually.
-  if (typeof target.constructor.proto === 'function') {
-    target.constructor.proto();
+  let superClass = Object.getPrototypeOf(target.constructor);
+
+  if (superClass.hasOwnProperty('proto') && typeof superClass.proto === 'function') {
```

Comment:
Definitely will add a test, but I do think this works as is. Once we hit the ES class chain, all the classes get eagerly evaluated, so actions should exist by the time the next class is being defined